Switch to Cura because MiracleGrue is abandoned

skalnik opened the pull request:
This is a WIP PR switching to CuraEngine, the slicer used in Ultimaker's Cura slicer.
### TODO
- [x] Remove grue from `vendor/` & replace with Cura Engine
- [x] Compile Cura with `vendor/Makefile`
- [x] Switch slicing to Cura
- [ ] Replace s3g
- [ ] Ensure it works on the Replicator 2 at GitHub HQ 3.0
- [ ] Get configuration stuff working

skalnik committed:
Start ripping out MiracleGrue :fire:

skalnik committed:
Add dat cura engine :car:

skalnik committed:
Make CuraEngine instead of grue :star:

sshirokov commented:
:zap:

skalnik commented:
Oh joy, Cura is configured with a `.cpp` file and then you have to recompile. That's a bit annoying. See https://github.com/Ultimaker/CuraEngine/blob/master/settings.cpp

skalnik committed:
Use cura for slicing :hocho:

sshirokov commented:
I have an [open PR](https://github.com/Ultimaker/CuraEngine/pull/55) against CuraEngine to add a config file.

**Update** In master

skalnik committed:
Bump cura :fist:

skalnik committed:
Use dat config :clipboard:

skalnik committed:
Setup a new default config

skalnik committed:
Align the rest of the file :point_right:

skalnik committed:
Keep this in here in case my computer explodes :boom:

Just a note of GCode issues

sshirokov committed:
Copy pasta start/endCode from s3g

Source: https://github.com/makerbot/s3g/blob/master/makerbot_driver/profiles/Replicator2.json

sshirokov committed:
Fix the endcode

skalnik committed:
Bump Cura :fist:

sshirokov commented on Makefile:
```diff
@@ -3,8 +3,8 @@ ROOT = $(shell pwd)
-GRUE ?= $(ROOT)/vendor/Miracle-Grue/bin/miracle_grue
-GRUE_CONFIG ?= default
+CURA ?= $(ROOT)/vendor/cura/CuraEngine
+CURA_CONFIG ?= default
```
We should probably name the config file something like "makerbot" instead of "default"

skalnik commented:
Some more fun news. S3G, says that G0 is unsupported GCode. However, running the same GCode through [GPX](https://github.com/whpthomas/GPX) works fine.

So to get cura working properly, we'll likely need to replace S3G as well, unless we feel like fucking with GCode manually before sending it to S3G.

skalnik committed:
Name the default config file specific to the rep2 :shrimp:

skalnik commented:
GPX doesn't want to stream to the Rep2 it seems :cry:

sshirokov commented:
I have a fix for GPX cooked and tested-ish. It's not perfect, but it makes it work.

https://github.com/sshirokov/GPX/compare/fix-the-stream

I'm also writing a similar tool that should be a bit more sane to deal with.
